usecases/ast_eval/evaluate_ast.go

```diff
@@ -19,19 +19,29 @@ func EvaluateAst(ctx context.Context, environment AstEvaluationEnvironment, node
 
 	childEvaluationFail := false
 
-	evalChild := func(child ast.Node) ast.NodeEvaluation {
+	// Only interested in lazy callback which will have default value if an error is returned
+	attrs, _ := node.Function.Attributes()
+
+	evalChild := func(child ast.Node) (childEval ast.NodeEvaluation, evalNext bool) {
 		childEval, ok := EvaluateAst(ctx, environment, child)
+
 		if !ok {
 			childEvaluationFail = true
+			return
```

Comment:
Should we stop evaluating at the first child returning an error, like this? There may be a good reason to continue evaluating other children if a previous one fails.